src/main/java/dev/ebrydeu/spring_boot_library/services/UserService.java

```diff
@@ -0,0 +1,76 @@
+package dev.ebrydeu.spring_boot_library.services;
+import dev.ebrydeu.spring_boot_library.domain.dto.UserDto;
+import dev.ebrydeu.spring_boot_library.domain.entities.User;
+import dev.ebrydeu.spring_boot_library.exception.CustomExceptions;
+import dev.ebrydeu.spring_boot_library.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+@Service
+public class UserService  {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+    public List<UserDto> getAllUsers() {
+        List<User> users = userRepository.findAll();
+        return users.stream()
+                .map(UserDto::map)
+                .collect(Collectors.toList());
+    }
+    public UserDto getUserById(Long userId) {
+        Optional<User> userOptional = userRepository.findById(userId);
+        if (userOptional.isEmpty()) {
+            throw new CustomExceptions.NotFoundException("User not found with id: " + userId);
+        }
```

Comment:
To maintain consistency, do the same as in `Message`, or alternatively, adjust `Message` to match the "User not found" convention.

```Java 
Message message = messageRepository.findById(id).orElseThrow(() -> new CustomExceptions.NotFoundException("Message not found with id: " + id));